In [1]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
Ionpos_Methanediol_ang_nc=[[ 1.35170 ,  -0.08370,    0.00000  ], 
                    [-0.52950,    1.24650 ,  -0.00000  ], 
                    [-0.06950,   -0.09830,   -0.00000  ],
                    [-0.43450 ,  -0.61450  ,  0.88910  ],
                    [-0.43450 ,  -0.61450 ,  -0.88910 ],
                    [1.62470 ,  -1.02540  ,  0.00000  ],
                    [-1.50830 ,   1.19000  ,  0.00000]]
ions_Methanediol=['O','O','C','H','H','H','H']

In [3]:
Ionpos_Methanediol_ang_nr= Ionpos_Methanediol_ang_nc-np.array([-0.43450 ,  -0.61450 ,  -0.88910])
print(Ionpos_Methanediol_ang_nr)

[[ 1.7862  0.5308  0.8891]
 [-0.095   1.861   0.8891]
 [ 0.365   0.5162  0.8891]
 [ 0.      0.      1.7782]
 [ 0.      0.      0.    ]
 [ 2.0592 -0.4109  0.8891]
 [-1.0738  1.8045  0.8891]]


In [4]:
b=np.mat([0,0,1])
a=np.mat([ 0.365 ,  0.5162 , 0.8891])
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Methanediol=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Methanediol_ang=np.array(np.dot(Rot_Methanediol,np.mat(Ionpos_Methanediol_ang_nr.T))).T

In [5]:
 Ionpos_Formaldehyde_ang_nc=np.array([[ 0.000000   ,  .000000   ,  .683501],
            [.000000   ,  .000000 ,   -.536614],
            [.000000  ,   .934390  , -1.124164],
            [.000000  ,  -.934390  , -1.124164]])
ions_Formaldehyde=['O','C','H','H']

In [6]:
Ionpos_Formaldehyde_ang_nr= Ionpos_Formaldehyde_ang_nc-np.array([.000000  ,  -.934390  , -1.124164])
print(Ionpos_Formaldehyde_ang_nr)

[[0.       0.93439  1.807665]
 [0.       0.93439  0.58755 ]
 [0.       1.86878  0.      ]
 [0.       0.       0.      ]]


In [7]:
b=np.mat([0,0,1])
a=np.mat([0.     ,  0.93439 , 0.58755 ])
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Formaldehyde=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Formaldehyde_ang=np.array(np.dot(Rot_Formaldehyde,np.mat(Ionpos_Formaldehyde_ang_nr.T))).T


In [8]:

Coords1=(np.linspace(0,0.5,2))/2
Coords2=(np.linspace(0,0.5,2))/2
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.   0.25] [0.   0.25] [-0.2 -0.1  0.   0.1  0.2]


In [9]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [10]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.250 -0.200 

Ion Pt 0.000 0.250 -0.100 

Ion Pt 0.000 0.250 0.000 

Ion Pt 0.000 0.250 0.100 

Ion Pt 0.000 0.250 0.200 

Ion Pt 0.250 0.000 -0.200 

Ion Pt 0.250 0.000 -0.100 

Ion Pt 0.250 0.000 0.000 

Ion Pt 0.250 0.000 0.100 

Ion Pt 0.250 0.000 0.200 

Ion Pt 0.250 0.250 -0.200 

Ion Pt 0.250 0.250 -0.100 

Ion Pt 0.250 0.250 0.000 

Ion Pt 0.250 0.250 0.100 

Ion Pt 0.250 0.250 0.200 



In [11]:
Coords1=(np.linspace(0,0.5,2))
Coords2=(np.linspace(0,0.5,2))
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.  0.5] [0.  0.5] [-0.2 -0.1  0.   0.1  0.2]


In [12]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [13]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.500 -0.200 

Ion Pt 0.000 0.500 -0.100 

Ion Pt 0.000 0.500 0.000 

Ion Pt 0.000 0.500 0.100 

Ion Pt 0.000 0.500 0.200 

Ion Pt 0.500 0.000 -0.200 

Ion Pt 0.500 0.000 -0.100 

Ion Pt 0.500 0.000 0.000 

Ion Pt 0.500 0.000 0.100 

Ion Pt 0.500 0.000 0.200 

Ion Pt 0.500 0.500 -0.200 

Ion Pt 0.500 0.500 -0.100 

Ion Pt 0.500 0.500 0.000 

Ion Pt 0.500 0.500 0.100 

Ion Pt 0.500 0.500 0.200 



In [14]:
for i in range(len(Ionpos_Formaldehyde_ang)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ang[i])):
        print("%.3f" % Ionpos_Formaldehyde_ang[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 



In [15]:
for i in range(len(Ionpos_Methanediol_ang)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ang[i])):
        print("%.3f" % Ionpos_Methanediol_ang[i][j], end = ' ')
    print('\n')

Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 



In [16]:
Ionpos_Formaldehyde_p=np.array(Ionpos_Formaldehyde_ang)-array(Ionpos_Formaldehyde_ang[-1])
Ionpos_Methanediol_p=np.array(Ionpos_Methanediol_ang)-array(Ionpos_Methanediol_ang[4])
for i in range(len(Ionpos_Formaldehyde_p)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_p[i])):
        print("%.3f" % Ionpos_Formaldehyde_p[i][j], end = ' ')
    print('\n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_p)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_p[i])):
        print("%.3f" % Ionpos_Methanediol_p[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 

__________________________
Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 



In [17]:
Ionpos_Formaldehyde_p=Ionpos_Formaldehyde_p #2.05 is the bond length of carbon and Pt
Ionpos_Methanediol_p=Ionpos_Methanediol_p

In [46]:
#conversion factors
a_Pt=0.392 #nm
a_Pt_ang=a_Pt*10 #angstrom
a_Pt_ang1=2*a_Pt_ang/sqrt(2) #angstrom
a_Pt_ang2=2*a_Pt_ang/sqrt(2)#angstrom
a_Pt_ang3=6*a_Pt_ang #angstrom

print(a_Pt_ang3,a_Pt_ang1*1.88973,a_Pt_ang2*1.88973,a_Pt_ang3*1.88973)
A100_Formaldehyde=np.array([np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang1,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang2,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang3])
A100_Ionpos_Methanediol=np.array([np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang1,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang2,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang3])

23.52 10.47612863727537 10.47612863727537 44.446449599999994


In [32]:
Ionpos_Formaldehyde_pp=Ionpos_Formaldehyde_p/A100_Formaldehyde.T

In [33]:
Ionpos_Methanediol_pp=Ionpos_Methanediol_p/A100_Ionpos_Methanediol.T

In [39]:
for i in range(len(Ionpos_Formaldehyde_pp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_pp[i])):
        print("%.3f" % Ionpos_Formaldehyde_pp[i][j], end = ' ')
    print('1 \n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_pp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_pp[i])):
        print("%.3f" % Ionpos_Methanediol_pp[i][j], end = ' ')
    print('1 \n')

ion O 0.000 -0.186 0.075 1 

ion C 0.000 0.000 0.047 1 

ion H 0.000 0.179 0.067 1 

ion H 0.000 0.000 0.000 1 

__________________________
ion O 0.240 -0.020 0.067 1 

ion O -0.099 0.220 0.067 1 

ion C 0.000 0.000 0.046 1 

ion H -0.107 -0.152 0.062 1 

ion H 0.000 0.000 0.000 1 

ion H 0.301 -0.173 0.052 1 

ion H -0.264 0.226 0.052 1 



In [47]:
#distance to platinum
Dis_ang=.186*10 #literature value in bohr
Dis=Dis_ang/(a_Pt_ang3) #in units of the lattice constants 


#summing the position of the central atom in the top most layer of  the surface

Dis

0.07908163265306122

In [48]:
Ionpos_Formaldehyde_ppp=Ionpos_Formaldehyde_pp+np.array([0,0,Dis-0.047])+np.array([0.5,0.5,0.1667])
Ionpos_Methanediol_ppp=Ionpos_Methanediol_pp+np.array([0,0,Dis-0.047  ])+np.array([0.5,0.5,0.1667])

In [49]:
for i in range(len(Ionpos_Formaldehyde_ppp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ppp[i])):
        print("%.3f" % Ionpos_Formaldehyde_ppp[i][j], end = ' ')
    print(' 1\n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_ppp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ppp[i])):
        print("%.3f" % Ionpos_Methanediol_ppp[i][j], end = ' ')
    print(' 1\n')

ion O 0.500 0.314 0.273  1

ion C 0.500 0.500 0.246  1

ion H 0.500 0.679 0.266  1

ion H 0.500 0.500 0.199  1

__________________________
ion O 0.740 0.480 0.266  1

ion O 0.401 0.720 0.266  1

ion C 0.500 0.500 0.245  1

ion H 0.393 0.348 0.260  1

ion H 0.500 0.500 0.199  1

ion H 0.801 0.327 0.251  1

ion H 0.236 0.726 0.251  1



In [25]:
sqrt(sum((Ionpos_Formaldehyde_ppp[1]-Ionpos_Formaldehyde_ppp[0])**2))*5*a_Pt_ang 

3.70910597084521

In [26]:
sqrt(sum((np.array(Ionpos_Formaldehyde_ang[1])-np.array(Ionpos_Formaldehyde_ang[0]))**2))

1.220115

ion Pt 0.000 0.000 -0.200 1
ion Pt 0.500 0.500 -0.200 1
ion Pt 0.000 0.50 -0.200 1
ion Pt 0.50 0.000 -0.200 1
ion Pt 0.750 0.250 -0.100 1 
ion Pt 0.250 0.750 -0.100 1
ion Pt 0.250 0.250 -0.100 1
ion Pt 0.750 0.750 -0.100 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.100 1 
ion Pt 0.250 0.750 0.100 1
ion Pt 0.250 0.250 0.100 1
ion Pt 0.750 0.750 0.100 1
ion Pt 0.000 0.000 0.200 1
ion Pt 0.000 0.50 0.200 1
ion Pt 0.50 0.000 0.200 1
ion Pt 0.50 0.50 0.200 1

In [57]:
#for carbon monoxide 
CO_dis=0.112*10/(a_Pt_ang3)
print(CO_dis)

0.04761904761904762


ion Pt 0.000 0.000 -0.1667 1
ion Pt 0.500 0.500 -0.1667 1
ion Pt 0.000 0.50 -0.1667 1
ion Pt 0.50 0.000 -0.1667 1
ion Pt 0.750 0.250 -0.0833 1
ion Pt 0.250 0.750 -0.0833 1
ion Pt 0.250 0.250 -0.0833 1
ion Pt 0.750 0.750 -0.0833 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1
ion Pt 0.000 0.50 0.000 1
ion Pt 0.500 0.50 0.000 1
ion Pt 0.750 0.250 0.0833 1
ion Pt 0.250 0.750 0.0833 1
ion Pt 0.250 0.250 0.0833 1
ion Pt 0.750 0.750 0.0833 1
ion Pt 0.000 0.000 0.1667 1
ion Pt 0.000 0.50 0.1667 1
ion Pt 0.50 0.000 0.1667 1
ion Pt 0.50 0.50 0.1667 1

In [54]:
#for carbon monoxide 
CO_dis=0.112*10/(a_Pt_ang3)
print(CO_dis)
print(CO_dis+0.246)

0.04761904761904762
0.2936190476190476
